<launch>
<!-- Launch file for Livox Horizon LiDAR -->
 
    <rosparam command="load" file="$(find fast_lio_localization)/config/config.yaml"/>

	<arg name="rviz" default="true" />

	<rosparam command="load" file="$(find fast_lio_localization)/config/mid360.yaml" />

	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="3"/>
	<param name="dense_publish_enable" type="bool" value="1" />
	<param name="filter_size_surf" type="double" value="1" />
	<param name="filter_size_map" type="double" value="1" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
	<param name="pcd_save_enable" type="bool" value="0" />
    <node pkg="fast_lio_localization" type="fastlio_mapping" name="laserMapping" output="screen"  />

	<param name="direct_map_path" type="string" value="$(find fast_lio)/PCD/map.pcd"/>
	<param name="max_len" type="int" value="10000"/>
	<param name="use_config_pose" type="int" value="0"/>


	<!-- loalization-->
	<node pkg="fast_lio_localization" type="direct_localization_mid360.py" name="global_localization" output="screen" />

	<!-- transform  fusion-->
	<node pkg="fast_lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen" />



	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization_move_base2.rviz" />
	</group>



	<include file="$(find pointcloud_to_laserscan)/launch/PointsCloud2toLaserscan.launch">
			<arg name="scan_name" value="/cloud_registered"/>
	</include>





</launch>